scripts/etc/init.d/rcS

```diff
@@ -85,6 +85,24 @@ sanitize_inputfile()
     fi
 }
 
+install_files()
+{
+  file_to_read=$1
+  sanitize_inputfile file_to_read
+  while read -r line; do
+    comment=$(echo "$line" | cut -c1)
```

Comment:
I think you should be able to replace the comment filtering with the following construct:
```
while read -r line; do
        <do stuff with line>
done < "${file_to_read}" | grep -v ^#
```